lib/src/backends/record_replay/proxy.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// An object that uses [noSuchMethod] to dynamically handle invocations
+/// (property getters, property setters, and method invocations).
+abstract class ProxyObject {}
```

Comment:
@proxy doesn't buy me anything w.r.t static analysis, since I explicitly implement the classes in question.  This is just to allow me to have a sensible type for `_proxyObject`...